### PR TITLE
Add initial database schema and middleware

### PR DESCRIPTION
**Pull request by jcamacho14**

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,8 @@
+DB_DRIVER=pg
+DB_HOST=localhost
+DB_NAME=srie
+DB_USER=srie
+DB_PASS=srie
+DB_PORT=5432
```
Comment (jcamacho14) [inline .env]: This isn't supposed to be commited, but it's the docker information, it is already commited on the docker-compse.yml, so I decided to include it.

Comment (yeco) [reply .env]: Yep. This definitely shouldn't be committed, you can rename it to .env.sample, though.

File: db/index.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
+const { Pool } = require("pg");
+
+const pool = new Pool({
+  user: "srie",
```
Comment (yeco) [inline db/index.js]: Please use ENV vars here.

File: db/migrations/20200322172041-initial-schema.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
+'use strict';
```
Comment (yeco) [inline db/migrations/20200322172041-initial-schema.js]: `dbmigrate` seems pretty nice, I've used `node-pg-migrate` which doesn't have as many options as this one . Although this seems pretty verbose for a migration, I assume this is needed only for the first one and then we can use `db-migrate create` from now on.

File: package.json [hunk 4]
```diff
@@ -3,13 +3,17 @@
   "version": "0.1.0",
   "private": true,
   "scripts": {
-    "dev": "next dev",
+    "dev": "db-migrate up && next dev",
     "build": "next build",
     "start": "next start"
   },
   "dependencies": {
+    "db-migrate": "^0.11.6",
```
Comment (yeco) [inline package.json]: Please consider including migration commands on the `README`.

File: pages/api/countries/[code].js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import nextConnect from "next-connect";
+import CountryService from "../../../services/Country.service";
+
+const handler = nextConnect();
+
+handler.get(async (req, res) => {
+  CountryService.getByCode(req.query.code).then(results => {
```
Comment (yeco) [inline pages/api/countries/[code].js]: 👌

File: db/index.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
+const { Pool } = require("pg");
```
Comment (yeco) [inline db/index.js]: Using `MassiveJS` could make your life easier, https://massivejs.org/. Up to you.

File: .env (hunk 1 above)
Comment (jcamacho14) [reply .env]: Ok, I can exclude it or rename it.

File: db/migrations/20200322172041-initial-schema.js (hunk 3 above)
Comment (jcamacho14) [reply db/migrations/20200322172041-initial-schema.js]: That file is generated by the tool on the `db-migrate create ...`, the only ones that are supposed to be changed are the corresponding SQL files. I chose this option to be able to use plain SQL. I took a look at the `node-pg-migrate`, but I didn't a way to use SQL, or at least, I found it faster on `db-migrate`

File: db/index.js (hunk 6 above)
Comment (jcamacho14) [reply db/index.js]: Do you know if I can use SQL while using MassiveJS? I would prefer to avoid specific ORM syntax

Comment (yeco) [reply db/index.js]: Yep, you can, Massive isn't a orm but a data mapper, so it doesn't feel intrusive.

File: db/migrations/20200322172041-initial-schema.js (hunk 3 above)
Comment (yeco) [reply db/migrations/20200322172041-initial-schema.js]: 👌

File: .env (hunk 1 above)
Comment (yeco) [reply .env]: Renaming is better, that way it stays as docu.

File: db/index.js (hunk 6 above)
Comment (jcamacho14) [reply db/index.js]: I will take a look. For now I updated the other requests

**Pull request merged**